comp/host-profiler/collector/impl/receiver/factory.go

```diff
@@ -20,10 +20,15 @@ import (
 	ebpfcollector "go.opentelemetry.io/ebpf-profiler/collector"
 )
 
+// GetFactoryName returns the name of the factory.
+func GetFactoryName() string {
```

Comment:
Nit: technically this is the component type, might be better to create a var `componentType = "hostprofiler"`